pkg/controller/operatingsystemconfig/actuator.go

```diff
@@ -54,13 +58,39 @@ func NewActuator(mgr manager.Manager, extensionConfig Config) operatingsystemcon
 
 func init() {
 	var err error
+	scheme := runtime.NewScheme()
+	runtimeutils.Must(configv1alpha1.AddToScheme(scheme))
+	decoder = serializer.NewCodecFactory(scheme).UniversalDecoder()
 	ntpConfigTemplate, err = template.New("ntp-config").Parse(ntpConfigTemplateContent)
 	if err != nil {
 		panic(fmt.Errorf("failed to parse NTP config template: %w", err))
 	}
 }
 
+func GetProviderConfiguration(osc *extensionsv1alpha1.OperatingSystemConfig) (*configv1alpha1.ExtensionConfig, error) {
+	if osc.Spec.ProviderConfig == nil {
+		return nil, nil
+	}
+
+	obj := &configv1alpha1.ExtensionConfig{}
+	if _, _, err := decoder.Decode(osc.Spec.ProviderConfig.Raw, nil, obj); err != nil {
+		return nil, fmt.Errorf("failed to decode provider config: %+v", err)
+	}
+
+	return obj, nil
+}
+
 func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, osc *extensionsv1alpha1.OperatingSystemConfig) ([]byte, []extensionsv1alpha1.Unit, []extensionsv1alpha1.File, *extensionsv1alpha1.InPlaceUpdatesStatus, error) {
+	// Try to load dynamic .spec.providerConfig in OSC
+	config, err := GetProviderConfiguration(osc)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+	if config != nil {
+		//Overwrite shoot-specific ExtensionConfig with global ExtensionConfig
+		a.extensionConfig.ExtensionConfig = config
```

Comment:
I think this will overwrite the `ExtensionConfig` for all clusters that are reconciled after this. As `a` and also `ExtensionConfig` are pointers.